criteria.c

```diff
@@ -368,7 +371,6 @@ struct mako_criteria *create_criteria_from_notification(
 	criteria->urgency = notif->urgency;
 	criteria->category = strdup(notif->category);
 	criteria->desktop_entry = strdup(notif->desktop_entry);
-	criteria->summary = strdup(notif->summary);
```

Comment:
This actually is used, when determining how to group notifications (see dbus/xdg.c:395, for example). Currently even with this change grouping will still require exact matches, since there's no way in the group option to specify a regex for this field. I think that's fine for the initial implementation, we may not ever want to get fancier than that.

Anyway, this is going to require creating a regex that matches exactly the literal summary string of the notification. I don't see an included function for escaping regex special characters in regex(3), which is unfortunate. We will either have to write our own, which needs to escape all of `.^$*+?()[{\|`, or hack around the problem by having a separate field and a flag that determines whether to compare using the regex field or the string field. Not really a fan of the hack.